samples/XAMLCatalog/XAMLCatalog/UIKitControls/UIButtonWithControlsViewController.m

```diff
@@ -0,0 +1,95 @@
+    // Set up the stage for the UIButton that will be manipulated
+    self.title = @"UIButton with controls";
+    self.view.backgroundColor = [UIColor whiteColor];
+    self.view.frame = CGRectMake(0.0f, 0.0f, 200.0f, 200.0f);
+
+    // Configure the button
+    [self setupButton];
+
+    // TableViewController hosting the controls that manipulate UIButton
+    _menuTVC = [[MenuTableViewController alloc] init];
+    _menuTVC.view.frame = CGRectMake(0.0f, 200.0f, 200.0f, 0.0f /* setting it to 0 allows vertical scrolling */);
+    _menuTVC.tableView.allowsSelection = NO;
+    [self.view addSubview:_button];
+    [self.view addSubview:_menuTVC.view];
+
+    // UITextField that changes the text on UIButton.titleLabel for normal state
+    _textButtonNormal = [[UITextField alloc] initWithFrame:CGRectMake(0.0f, 0.0f, 150.0f, 40.0f)];
+    _textButtonNormal.textColor = [UIColor blackColor];
+    _textButtonNormal.backgroundColor = [UIColor lightGrayColor];
+    _textButtonNormal.textAlignment = UITextAlignmentCenter;
+    _textButtonNormal.contentVerticalAlignment = UIControlContentVerticalAlignmentCenter;
+    _textButtonNormal.borderStyle = UITextBorderStyleLine;
+    _textButtonNormal.font = [UIFont systemFontOfSize:15.0f];
+    _textButtonNormal.adjustsFontSizeToFitWidth = YES;
+    _textButtonNormal.delegate = self;
+    [_textButtonNormal addTarget:self action:@selector(onTextChanged:) forControlEvents:UIControlEventEditingChanged];
+    [_menuTVC addMenuItemView:_textButtonNormal andTitle:@"Text - normal state"];
+
+    // UISlider that changes the text color on UIButton titleLabel for normal state
+    _sliderTitleColorNormal = [[UISlider alloc] initWithFrame:CGRectMake(0.0f, 0.0f, 100.0f, 40.0f)];
+    _sliderTitleColorNormal.minimumValue = 0.0f;
+    _sliderTitleColorNormal.maximumValue = 360.0f;
+    _sliderTitleColorNormal.continuous = YES;
+    _sliderTitleColorNormal.value = 0.0f;
+    [_sliderTitleColorNormal addTarget:self action:@selector(titleColorNormalValueChanged) forControlEvents:UIControlEventValueChanged];
+    [_menuTVC addMenuItemView:_sliderTitleColorNormal andTitle:@"Background color - normal state"];
+}
+
+// Initial configuration parameters for the UIButton
+- (void)setupButton {
+    _button = [[UIButton alloc] initWithFrame:self.view.bounds];
+    [_button setTitle:@"Button Normal" forState:UIControlStateNormal];
+    [_button setTitle:@"Button Highlighted" forState:UIControlStateHighlighted];
+    [_button setTitleColor:[UIColor blackColor] forState:UIControlStateNormal];
+    [_button setTitleColor:[UIColor blueColor] forState:UIControlStateHighlighted];
+
+    _button.backgroundColor = [UIColor lightGrayColor];
+}
+
+// Delegate to set the UIButton.titleLabel text
+- (void)onTextChanged:(UITextField*)textField {
+    [_button setTitle:textField.text forState:UIControlStateNormal];
+}
+
+// Delegate to dismiss the keyboard
+- (BOOL)textFieldShouldReturn:(UITextField*)textField {
+    [textField resignFirstResponder];
```

Comment:
apparently you have to do it youself. nvm

---
In reply to: [97840037](https://github.com/Microsoft/WinObjC/pull/1771#discussion_r97840037) [](ancestors = 97840037,97799074,97699783)